components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/mediators/ClaimBasedResourceAccessValidationMediator.java

```diff
@@ -62,42 +62,45 @@ public boolean mediate(MessageContext messageContext) {
 
         claimValueSentInToken = jwtTokenClaims.get(accessVerificationClaim);
 
-        if (StringUtils.isBlank(claimValueSentInToken)) {
-            log.error("The configured resource access validation claim is " +
-                    "not present in the token.");
-            handleFailure(HttpStatus.SC_FORBIDDEN, messageContext, String.format("Token doesn't contain the " +
-                    "claim \"%s\"", accessVerificationClaim), null);
-            return false;
-        }
-
-        if (StringUtils.isNotBlank(accessVerificationClaimValueRegex)) {
-            log.debug("A regex is provided, hence, validating the claim values using the provided regex.");
-            Pattern pattern = Pattern.compile(accessVerificationClaimValueRegex);
-            Matcher configuredClaimValueMatcher = pattern.matcher(accessVerificationClaimValue);
-            Matcher tokenSentClaimValueMatcher = pattern.matcher(claimValueSentInToken);
-
-            if ((configuredClaimValueMatcher.matches() && tokenSentClaimValueMatcher.matches())
-                    || shouldAllowValidation) {
-                log.debug("Claim values match or the flow is configured to allow when claims doesn't match. " +
-                        "Hence the flow is allowed.");
-                return true;
-            } else {
-                log.debug("Claim values don't match. Hence the flow is not allowed.");
-                handleFailure(HttpStatus.SC_FORBIDDEN, messageContext, CLAIMS_MISMATCH_ERROR_MSG, null);
-                return false;
+        try {
+            if (StringUtils.isBlank(claimValueSentInToken)) {
+                log.error("The configured resource access validation claim is " +
+                        "not present in the token.");
+                throw new APISecurityException(APISecurityConstants.API_AUTH_ACCESS_TOKEN_CLAIMS_INVALID,APISecurityConstants.API_AUTH_ACCESS_TOKEN_CLAIMS_INVALID_MESSAGE, String.format("Token doesn't contain the " +
+                        "claim \"%s\"", accessVerificationClaim));
             }
-        } else {
-            log.debug("A regex is not provided, validating the claim values based on equality.");
-            if ((StringUtils.equals(accessVerificationClaimValue, claimValueSentInToken)) || shouldAllowValidation) {
-                log.debug("Claim values match or the flow is configured to allow when claims doesn't match. " +
-                        "Hence the flow is allowed.");
-                return true;
+
+            if (StringUtils.isNotBlank(accessVerificationClaimValueRegex)) {
```

Comment:
Shall we replace the if and else blocks to improve readability

Reply:
I put the else block to improve the readability.